lib/remote-connector.js

```diff
@@ -104,16 +108,48 @@ function createProxyMethod(Model, remotes, remoteMethod) {
       callback = utils.createPromiseCallback();
     }
 
-    var reqOptions
-    var lastArg = args[args.length - 1]
-    if (
-      typeof lastArg === 'object' &&
-      lastArg &&
-      'reqOptions' in lastArg &&
-      typeof lastArg.reqOptions === 'object' &&
-      lastArg.reqOptions
-    ) {
-      reqOptions = args.pop().reqOptions
+    // Support taking reqOptions off the callback, or off the last arg (after the callback)
+    // which is commonly an options object - this allows it to be passed without changing the
+    // method call signature, which might have unintended consequences when a remote method is
+    // invoked in a non-remote context, especially for internal loopback methods
+    var reqOptions;
+    if (lastArgIsFunc) {
+      if (typeof callback[reqOptionsSymbol] === 'object' && callback[reqOptionsSymbol]) {
+        reqOptions = callback[reqOptionsSymbol];
+        delete callback[reqOptionsSymbol];
+      }
+    }
+
+    // If reqOptions was not set as a symbol on the callback, it can be set as a symbol on the
+    // last object-arg of the called function. In this case we can extract the reqOptions off
+    // the object and delete the property
+    if (!reqOptions) {
+      var lastArg = args[args.length - 1];
+      if (
+        typeof lastArg === 'object' &&
+        lastArg &&
+        typeof lastArg[reqOptionsSymbol] === 'object' &&
+        lastArg[reqOptionsSymbol]
+      ) {
+        reqOptions = lastArg[reqOptionsSymbol]
+        delete lastArg[reqOptionsSymbol];
```

Comment:
If you conditionally remove the last argument if its an empty object then you don't need the following code block. You may however want to add a comment

```suggestion
        delete lastArg[reqOptionsSymbol];
        
        if (Object.keys(lastArg).length === 0) {
          args.pop()
        }
```